src/plugins/intel_npu/src/plugin/npuw/just_sync_infer_request.cpp

```diff
@@ -193,7 +194,7 @@ ov::npuw::JustInferRequest::JustInferRequest(const std::shared_ptr<ov::npuw::Com
 
             // No update required to this tensor in runtime - so it can be set only once
             if (!comp_model_desc.update_required[cidx]) {
-                request->set_tensor(iport, ov::get_tensor_impl(closure));
+                request->set_tensor(iport, ov::get_tensor_impl(wbank->get(closure, *comp_model_desc.device_it)));
```

Comment:
In general case, you should take the `device_it` from `m_compiled_models[real_idx]`.

It works for you here as you only process the CWAI subgraphs here (where idx == real_idx), but in general case it is a trap we may fall into.

Reply:
Done